vernac/classes.ml

```diff
@@ -152,6 +152,40 @@ let default_locality () =
 let instance_locality =
   Attributes.hint_locality ~default:default_locality
 
+type event =
+  | NewClass of typeclass
+  | NewInstance of {
+      class_name : GlobRef.t;
+      instance : GlobRef.t;
+      info : Typeclasses.hint_info;
+      locality : Hints.hint_locality;
+    }
+
+let observers = ref CString.Map.empty
+
+let add_observer ~name o =
```

Comment:
Do we really need the name? We could just put the functions in a list instead of String.Map.

It would mean we can't put the name in ObserverError but I don't know if that matters. What is that exception wrapping even for? Maybe we should just let the exceptions through unchanged.

Reply:
I thought the name was informative: if plugin foobar adds a buggy observer, the name gives a hint about where to look.

But I really don't care.

Reply:
A really buggy observer would produce anomalies which we don't wrap though.

Meanwhile an observer may want to raise normal errors (eg some sort of linter), then the wrapping is just going to produce noise.

Reply:
Fair enough, I'm removing it.

Reply:
done